src/core/audio/AudioJack.cpp

```diff
@@ -286,11 +298,17 @@ void AudioJack::renamePort(AudioPort* port)
 }
 
 
+int AudioJack::setBufferSizeCallback(jack_nframes_t nframes, void* udata)
+{
+	auto thisClass = static_cast<AudioJack*>(udata);
+	delete[] thisClass->m_inputFrameBuffer;
+	thisClass->m_inputFrameBuffer = new jack_default_audio_sample_t[thisClass->channels() * nframes];
```

Comment:
Consider to use an `std::vector<jack_default_audio_sample_t>` that's simply resized in this block. This would also spare you the manual deletion in the destructor and would thus make the code a bit more memory safe.